File: java-tests/com/hounshell/kotlin-sonnet/java/TypeReferenceTests.java
package com.hounshell.kotlin_sonnet.java;

import static com.google.common.truth.Truth.assertThat;

import com.hounshell.kotlin_sonnet.TypeReference;

import org.junit.Test;

public class TypeReferenceTests {
    @Test
    public void typeReferenceByName() {
        assertThat(new TypeReference("foo.Bar").getFullyQualifiedName()).isEqualTo("foo.Bar");
    }

    @Test
    public void TypeReferenceByJavaClass() {
        assertThat(new TypeReference(String.class).getFullyQualifiedName()).isEqualTo(String.class.getCanonicalName());
    }
}

File: java-tests/com/hounshell/kotlin-sonnet/kotlin/TypeReferenceTests.kt
package com.hounshell.kotlin_sonnet.kotlin

import com.google.common.truth.Truth.assertThat
import com.hounshell.kotlin_sonnet.TypeReference

import org.junit.Test

class TypeReferenceTests {
    @Test
    fun `TypeReference by name`() {
        assertThat(TypeReference("foo.Bar").fullyQualifiedName).isEqualTo("foo.Bar")
    }

    @Test
    fun `TypeReference by java class`() {
        assertThat(TypeReference(String::class.java).fullyQualifiedName).isEqualTo(String::class.java.canonicalName)
    }
}
